templates/macros/events.html

```diff
@@ -314,4 +317,265 @@ <h2 class="text-center">No events {% if archive %}finished{% endif %} yet, pleas
     }
   }
 </script>
+{% endmacro %}
+
+{% macro filter_controls() %}
+<div class="vstack d-none" id="filter-controls">
+  <h2>Filter Events</h2>
+  <h4  id="category-title" class="d-none">Category</h4>
+  <div id="category-filters" class="d-none"></div>
+  <h4  id="tag-title" class="d-none mt-2">Tags</h4>
+  <div id="tag-filters" class="d-none"></div>
+  <hr>
+</div>
+{% endmacro %}
+
+{% macro filter_script() %}
+<script>
+  // these 'categories' are actually specified by the 'colour'
+  // property, but they fit well enough to categorise events
+  // and are probably more useful than the tags
+  // taken from colours.json
+  const ALLOWED_CATEGORIES = [
+      "gaming", "academic", "tech", "social", "inclusivity",
+      "publicity", "compcafe", "milk", "other"
```

Comment:
other category never seems to appear?
publicity also don't have a specifc event associated with them